Quarkus CLI version 999-SNAPSHOT and 3.19.2 fails to update applications on Windows

### Describe the bug

I use Quarkus CLI to update my applications, on Windows, my applications can't be updated with Quarkus CLI v999-SNAPSHOT, but it works with other Quarkus CLI versions like 3.19.1. Applications seems to be updated, but whole process fails anyway. See also https://github.com/quarkusio/quarkus/pull/46308.

### Expected behavior

No failures, no errors, same as with older Quarkus CLI versions.

### Actual behavior

Update results in following failure:

```
PS C:\cygwin\home\hudson\app> C:\cygwin\home\hudson\quarkus-test-suite\quarkus-dev-cli.bat update
[INFO] Error stacktraces are turned on. 
[INFO] Scanning for projects... 
[INFO]  
[INFO] ----------------------------< org.acme:app >----------------------------
[INFO] Building app 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- quarkus:999-SNAPSHOT:update (default-cli) @ app ---
[WARNING] quarkus:update goal is experimental, its options and output might change in future versions 
[INFO] Looking for the newly published extensions in registry.quarkus.io 
[INFO] Detected project Java version: 21 
[INFO]  
[INFO] ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[INFO]
[INFO] Suggested Quarkus platform BOM updates:
[INFO]  ~       io.quarkus.platform:quarkus-bom:pom:[3.15.3.1 -> 3.19.1]
[INFO]
[INFO] Suggested extensions updates for 'io.quarkus.platform:quarkus-bom':
[INFO]  v       io.quarkus:quarkus-arc (synced with BOM)
[INFO]  v       io.quarkus:quarkus-rest (synced with BOM)
[INFO]
[INFO]
[INFO] ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[INFO]
[INFO]
[INFO] Resolving recipes from 'io.quarkus:quarkus-update-recipes':
[INFO] * matching recipes directory 'core' found: 
[INFO]     - '3.18.alpha1.yaml' (3.15.3.1 -> 3.19.1)
[INFO]     - '3.19.alpha1.yaml' (3.15.3.1 -> 3.19.1)
[INFO] => 2 specific recipe(s) found
[INFO]  
[INFO] We have generated a recipe file to update your project (versions updates + specific recipes):
[INFO] target\rewrite\rewrite.yaml
[INFO] 

Do you want run the generated update recipe with OpenRewrite ? ([y]es, [n]o, [d]ry-run)

y 
[INFO] Running recipe with OpenRewrite 6.2.0... 
[INFO]
[INFO]  
[INFO]
[INFO]
[INFO]  ------------------------------------------------------------------------
[INFO] Executing (this may take a while):
C:\cygwin\home\hudson\app\mvnw.cmd -B -e org.openrewrite.maven:rewrite-maven-plugin:6.2.0:run -Drewrite.plainTextMasks=**/*.adoc,**/src/test/resources/__snapshots__/**/*.java,**/*.kt,*
*/*.md,**/src/main/codestarts/**/*.java,**/*.txt,**/META-INF/services/** -Drewrite.configLocation=C:\cygwin\home\hudson\app\target\rewrite\rewrite.yaml -Drewrite.recipeArtifactCoordina
tes=io.quarkus:quarkus-update-recipes:1.2.0 -Drewrite.activeRecipes=io.quarkus.openrewrite.Quarkus -Drewrite.pomCacheEnabled=false
[INFO]
[INFO] Rewrite process completed. Logs can be found at: target\rewrite\rewrite.log 
[INFO]  
[INFO]  
[INFO]
[INFO]
[INFO]  ------------------------------------------------------------------------
[INFO] Executing (this may take a while):
C:\cygwin\home\hudson\app\mvnw.cmd -B clean process-sources
[INFO]
[INFO] ------------------------------------------------------------------------ 
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  59.790 s
[INFO] Finished at: 2025-03-05T17:59:54+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:update (default-cli) on project app: The command to update the project exited with an errorLogs can be found
 at: target\rewrite\rewrite.log -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:update (default-cli) on project app: The command to update t
he project exited with an errorLogs can be found at: target\rewrite\rewrite.log
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101) 
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
PS C:\cygwin\home\hudson\app>
```

### How to Reproduce?

Steps to reproduce the behavior:

```
# use PowerShell to prepare Quarkus CLI v 999-SNAPSHOT
$quarkusCliFileContent = @"
@ECHO OFF
java -jar c:\\users\\hudson\\.m2\\repository\\io\\quarkus\\quarkus-cli\\999-SNAPSHOT\\quarkus-cli-999-SNAPSHOT-runner.jar %*
"@
New-Item -Path "$(pwd)\quarkus-dev-cli.bat" -ItemType File
Set-Content -Path "$(pwd)\quarkus-dev-cli.bat" -Value $quarkusCliFileContent
./quarkus-dev-cli.bat version

# now run tests

# attempt 1 - fails
C:\cygwin\home\hudson\quarkus-test-suite\quarkus-dev-cli.bat create app app --stream=3.2
cd app
C:\cygwin\home\hudson\quarkus-test-suite\quarkus-dev-cli.bat update -y --stream=3.15

# attempt 2 - fails
C:\cygwin\home\hudson\quarkus-test-suite\quarkus-dev-cli.bat create app app --stream=3.15
cd app
C:\cygwin\home\hudson\quarkus-test-suite\quarkus-dev-cli.bat update -y

# attempt 3 - fails
C:\cygwin\home\hudson\quarkus-test-suite\quarkus-dev-cli.bat create app app --stream=3.15
cd app
C:\cygwin\home\hudson\quarkus-test-suite\quarkus-dev-cli.bat update

# Quarkus CLI 3.19.1 -> pass
quarkus create app app --stream=3.2
cd app
quarkus update --stream=3.15
```

Or you can see failures if you use the jar directly:

```
java -jar c:\\users\\hudson\\.m2\\repository\\io\\quarkus\\quarkus-cli\\999-SNAPSHOT\\quarkus-cli-999-SNAPSHOT-runner.jar create app app --stream=3.15

java -jar c:\\users\\hudson\\.m2\\repository\\io\\quarkus\\quarkus-cli\\999-SNAPSHOT\\quarkus-cli-999-SNAPSHOT-runner.jar update
```


### Output of `uname -a` or `ver`

Windows Server 2019, Microsoft Windows Server 2022

### Output of `java -version`

Temurin 21

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.9.9

### Additional information

cc @ia3andy

Sorry, I got stuck last night when tried to reproduce it because I keep getting:

```
PS C:\cygwin\home\hudson\code-with-quarkus> java -jar c:\\users\\hudson\\.m2\\repository\\io\\quarkus\\quarkus-cli\\999-SNAPSHOT\\quarkus-cli-999-SNAPSHOT-runner.jar update
[ERROR]  Unable to run Quarkus project update : Failed to resolve extension catalog for io.quarkus:quarkus-bom:pom:999-SNAPSHOT. Make sure the groupId, artifactId and version 
are spelled correctly and the relevant Maven repositories are configured. 
```

it's definitely something I have seen before, maybe I didn't build Quarkus properly or something, I am looking for the step I have missed.

So the first error I got is `Caused by: org.codehaus.plexus.compiler.CompilerException: error: release version 21 not supported`. I'll post next comment with different error.

Logs for this one:

```
[INFO] Update in progress (this may take a while):
[INFO]   - executing OpenRewrite 6.2.0 command...
[INFO]  
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  35.446 s 
[INFO] Finished at: 2025-03-12T20:34:07+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:update (default-cli) on project app: The OpenRewrite 6.2.0 command exited with an error. Logs can be found at: target\rewrite\rewrite.log -> [Help 1] 
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:update (default-cli) on project app: The OpenRewrite 6.2.0 command exited with an error. Logs can be found at: tar
get\rewrite\rewrite.log
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159) 
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77) 
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
Caused by: org.apache.maven.plugin.MojoExecutionException: The OpenRewrite 6.2.0 command exited with an error. Logs can be found at: target\rewrite\rewrite.log
    at io.quarkus.maven.UpdateMojo.processProjectState (UpdateMojo.java:155)
    at io.quarkus.maven.QuarkusProjectStateMojoBase.doExecute (QuarkusProjectStateMojoBase.java:59)
    at io.quarkus.maven.QuarkusProjectMojoBase.execute (QuarkusProjectMojoBase.java:114)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

```

```
Running: C:\cygwin\home\hudson\app\mvnw.cmd -B -e org.openrewrite.maven:rewrite-maven-plugin:6.2.0:run -Drewrite.plainTextMasks=**/META-INF/services/**,**/*.txt,**/src/main/codestarts/**/*.java,**/*.md,**/*.kt,**/src/test/resources/__snap
shots__/**/*.java,**/*.adoc -Drewrite.configLocation=C:\cygwin\home\hudson\app\target\rewrite\rewrite.yaml -Drewrite.recipeArtifactCoordinates=io.quarkus:quarkus-update-recipes:1.2.0 -Drewrite.activeRecipes=io.quarkus.openrewrite.Quarkus 
-Drewrite.pomCacheEnabled=false
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/openrewrite/maven/rewrite-maven-plugin/6.2.0/rewrite-maven-plugin-6.2.0.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/openrewrite/maven/rewrite-maven-plugin/6.2.0/rewrite-maven-plugin-6.2.0.pom (27 kB at 53 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/com/fasterxml/jackson/jackson-bom/2.17.3/jackson-bom-2.17.3.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/com/fasterxml/jackson/jackson-bom/2.17.3/jackson-bom-2.17.3.pom (19 kB at 603 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/assertj/assertj-bom/3.27.3/assertj-bom-3.27.3.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/assertj/assertj-bom/3.27.3/assertj-bom-3.27.3.pom (3.7 kB at 118 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/rsocket/rsocket-bom/1.1.5/rsocket-bom-1.1.5.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/rsocket/rsocket-bom/1.1.5/rsocket-bom-1.1.5.pom (2.4 kB at 149 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/openrewrite/rewrite-bom/8.47.1/rewrite-bom-8.47.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/openrewrite/rewrite-bom/8.47.1/rewrite-bom-8.47.1.pom (6.1 kB at 196 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/openrewrite/maven/rewrite-maven-plugin/6.2.0/rewrite-maven-plugin-6.2.0.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/openrewrite/maven/rewrite-maven-plugin/6.2.0/rewrite-maven-plugin-6.2.0.jar (112 kB at 1.8 MB/s)
[INFO]
[INFO] ----------------------------< org.acme:app >----------------------------
[INFO] Building app 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-failsafe-plugin/3.3.1/maven-failsafe-plugin-3.3.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-failsafe-plugin/3.3.1/maven-failsafe-plugin-3.3.1.pom (11 kB at 366 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire/3.3.1/surefire-3.3.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire/3.3.1/surefire-3.3.1.pom (21 kB at 1.3 MB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-failsafe-plugin/3.3.1/maven-failsafe-plugin-3.3.1.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-failsafe-plugin/3.3.1/maven-failsafe-plugin-3.3.1.jar (56 kB at 3.5 MB/s)
[INFO]
[INFO] >>> rewrite:6.2.0:run (default-cli) > process-test-classes @ app >>>
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-junit5/3.15.3.1/quarkus-junit5-3.15.3.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-junit5/3.15.3.1/quarkus-junit5-3.15.3.1.pom (2.4 kB at 76 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-test-framework/3.15.3.1/quarkus-test-framework-3.15.3.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-test-framework/3.15.3.1/quarkus-test-framework-3.15.3.1.pom (3.2 kB at 102 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-test-common/3.15.3.1/quarkus-test-common-3.15.3.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-test-common/3.15.3.1/quarkus-test-common-3.15.3.1.pom (2.1 kB at 66 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-bootstrap-gradle-resolver/3.15.3.1/quarkus-bootstrap-gradle-resolver-3.15.3.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-bootstrap-gradle-resolver/3.15.3.1/quarkus-bootstrap-gradle-resolver-3.15.3.1.pom (4.5 kB at 281 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-junit5-properties/3.15.3.1/quarkus-junit5-properties-3.15.3.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-junit5-properties/3.15.3.1/quarkus-junit5-properties-3.15.3.1.pom (943 B at 30 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/jboss/marshalling/jboss-marshalling/2.2.1.Final/jboss-marshalling-2.2.1.Final.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/jboss/marshalling/jboss-marshalling/2.2.1.Final/jboss-marshalling-2.2.1.Final.pom (3.6 kB at 116 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/jboss/marshalling/jboss-marshalling-parent/2.2.1.Final/jboss-marshalling-parent-2.2.1.Final.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/jboss/marshalling/jboss-marshalling-parent/2.2.1.Final/jboss-marshalling-parent-2.2.1.Final.pom (5.1 kB at 339 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/rest-assured/rest-assured/5.5.0/rest-assured-5.5.0.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/rest-assured/rest-assured/5.5.0/rest-assured-5.5.0.pom (7.6 kB at 473 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/rest-assured/rest-assured-parent/5.5.0/rest-assured-parent-5.5.0.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/rest-assured/rest-assured-parent/5.5.0/rest-assured-parent-5.5.0.pom (22 kB at 704 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/rest-assured/json-path/5.5.0/json-path-5.5.0.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/rest-assured/json-path/5.5.0/json-path-5.5.0.pom (4.9 kB at 306 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/rest-assured/rest-assured-common/5.5.0/rest-assured-common-5.5.0.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/rest-assured/rest-assured-common/5.5.0/rest-assured-common-5.5.0.pom (3.2 kB at 213 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/rest-assured/xml-path/5.5.0/xml-path-5.5.0.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/rest-assured/xml-path/5.5.0/xml-path-5.5.0.pom (4.3 kB at 288 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-junit5/3.15.3.1/quarkus-junit5-3.15.3.1.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-junit5/3.15.3.1/quarkus-junit5-3.15.3.1.jar (178 kB at 3.8 MB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-test-common/3.15.3.1/quarkus-test-common-3.15.3.1.jar
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-bootstrap-gradle-resolver/3.15.3.1/quarkus-bootstrap-gradle-resolver-3.15.3.1.jar
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-junit5-properties/3.15.3.1/quarkus-junit5-properties-3.15.3.1.jar
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/jboss/marshalling/jboss-marshalling/2.2.1.Final/jboss-marshalling-2.2.1.Final.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-test-common/3.15.3.1/quarkus-test-common-3.15.3.1.jar (121 kB at 2.6 MB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/rest-assured/json-path/5.5.0/json-path-5.5.0.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/rest-assured/json-path/5.5.0/json-path-5.5.0.jar (60 kB at 763 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/rest-assured/rest-assured-common/5.5.0/rest-assured-common-5.5.0.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/rest-assured/rest-assured-common/5.5.0/rest-assured-common-5.5.0.jar (52 kB at 475 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/rest-assured/xml-path/5.5.0/xml-path-5.5.0.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/rest-assured/xml-path/5.5.0/xml-path-5.5.0.jar (81 kB at 646 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/io/rest-assured/rest-assured/5.5.0/rest-assured-5.5.0.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-junit5-properties/3.15.3.1/quarkus-junit5-properties-3.15.3.1.jar (6.4 kB at 29 kB/s) 
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/jboss/marshalling/jboss-marshalling/2.2.1.Final/jboss-marshalling-2.2.1.Final.jar (236 kB at 756 kB/s)
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/rest-assured/rest-assured/5.5.0/rest-assured-5.5.0.jar (747 kB at 2.0 MB/s)
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/io/quarkus/quarkus-bootstrap-gradle-resolver/3.15.3.1/quarkus-bootstrap-gradle-resolver-3.15.3.1.jar (3.0 MB at 8.1 MB/s)
[INFO]
[INFO] --- resources:3.3.1:resources (default-resources) @ app ---
[INFO] Copying 1 resource from src\main\resources to target\classes
[INFO]
[INFO] --- quarkus:3.15.3.1:generate-code (default) @ app ---
[INFO]
[INFO] --- compiler:3.13.0:compile (default-compile) @ app ---
[INFO] Recompiling the module because of changed source code.
[INFO] Compiling 1 source file with javac [debug parameters release 21] to target\classes
[INFO] ------------------------------------------------------------------------

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:compile (default-compile) on project app: Fatal error compiling: error: release version 21 not supported -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:compile (default-compile) on project app: Fatal error compiling
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.MojoExecutor.executeForkedExecutions (MojoExecutor.java:448)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:311)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162) 
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
Caused by: org.apache.maven.plugin.MojoExecutionException: Fatal error compiling
    at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:1231)
    at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:215)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174) 
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.MojoExecutor.executeForkedExecutions (MojoExecutor.java:448)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:311)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77) 
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
Caused by: org.codehaus.plexus.compiler.CompilerException: error: release version 21 not supported
    at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess (JavaxToolsCompiler.java:179)
    at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile (JavacCompiler.java:214)
    at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:1228)
    at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:215)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.MojoExecutor.executeForkedExecutions (MojoExecutor.java:448)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:311)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101) 
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
Caused by: java.lang.IllegalArgumentException: error: release version 21 not supported
    at com.sun.tools.javac.main.Arguments.reportDiag (Arguments.java:889)
    at com.sun.tools.javac.main.Arguments.handleReleaseOptions (Arguments.java:311)
    at com.sun.tools.javac.main.Arguments.processArgs (Arguments.java:350)
    at com.sun.tools.javac.main.Arguments.init (Arguments.java:246)
    at com.sun.tools.javac.api.JavacTool.getTask (JavacTool.java:191)
    at com.sun.tools.javac.api.JavacTool.getTask (JavacTool.java:119)
    at com.sun.tools.javac.api.JavacTool.getTask (JavacTool.java:68)
    at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess (JavaxToolsCompiler.java:125)
    at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile (JavacCompiler.java:214)
    at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:1228)
    at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:215)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328) 
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.MojoExecutor.executeForkedExecutions (MojoExecutor.java:448)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:311)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361) 
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

then I did `sed -i -e 's|<maven.compiler.release>21</maven.compiler.release>|<maven.compiler.release>17</maven.compiler.release>|' pom.xml` which resulted in:

```
NFO] ----------------------------< org.acme:app >----------------------------
[INFO] Building app 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- clean:3.2.0:clean (default-clean) @ app ---
[INFO] Deleting C:\cygwin\home\hudson\app\target
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  7.612 s
[INFO] Finished at: 2025-03-12T20:44:37+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-clean-plugin:3.2.0:clean (default-clean) on project app: Failed to clean project: Failed to delete C:\cygwin\home\hudson\app\target\rewrite\rewrite.log -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-clean-plugin:3.2.0:clean (default-clean) on project app: Failed to clean project: Failed to delete C:\cygwin\home\hudson\app\tar
get\rewrite\rewrite.log
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method) 
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to clean project: Failed to delete C:\cygwin\home\hudson\app\target\rewrite\rewrite.log
    at org.apache.maven.plugins.clean.CleanMojo.execute (CleanMojo.java:288)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568) 
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
Caused by: java.io.IOException: Failed to delete C:\cygwin\home\hudson\app\target\rewrite\rewrite.log
    at org.apache.maven.plugins.clean.Cleaner.delete (Cleaner.java:348)
    at org.apache.maven.plugins.clean.Cleaner.delete (Cleaner.java:290)
    at org.apache.maven.plugins.clean.Cleaner.delete (Cleaner.java:257)
    at org.apache.maven.plugins.clean.Cleaner.delete (Cleaner.java:257)
    at org.apache.maven.plugins.clean.Cleaner.delete (Cleaner.java:137)
    at org.apache.maven.plugins.clean.CleanMojo.execute (CleanMojo.java:267)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
[ERROR][ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR][ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

So when the release target is 17, it seems like error is `Caused by: java.io.IOException: Failed to delete C:\cygwin\home\hudson\app\target\rewrite\rewrite.log`

I was looking into this too :) I used quarkus-cli-999-20250312.022624-1379-runner.jar from sonatype snapshots repo.

In my case, the update command fails with `Caused by: java.io.IOException: Failed to delete C:\Users\Svobodovi\Downloads\app4\target\rewrite\rewrite.log` when running process-sources command: `c:\Users\Svobodovi\Downloads\app4\mvnw.cmd -B -e clean process-sources`

Log: [rewrite.log](https://github.com/user-attachments/files/19217229/rewrite.log)

`clean` target wants to delete target directory but it has active file log on rewrite.log file. Windows/NTFS has stricter file system access than Linux or macOS. 

To resolve this issue I think `rewrite.log` needs to be outside target directory or `clean` target should be dropped from the process-sources command. 


Used commands:
```
java -jar c:\Users\Svobodovi\Downloads\quarkus-cli-999-20250312.022624-1379-runner.jar create app app --stream=3.15
cd app
java -jar c:\Users\Svobodovi\Downloads\quarkus-cli-999-20250312.022624-1379-runner.jar update -y
```

Quarkus 3.19.2 has also the same behavior as described in this issue. The last working update on Windows is with Quarkus 3.19.1
```
java -jar c:\Users\Svobodovi\Downloads\quarkus-cli-3.19.2-runner.jar create app app --stream=3.15
cd app
java -jar c:\Users\Svobodovi\Downloads\quarkus-cli-3.19.2-runner.jar update -y
```

With 999-SNAPSHOT based CLI I also hit `[ERROR]  Unable to run Quarkus project update : Failed to resolve extension catalog for io.quarkus:quarkus-bom:pom:999-SNAPSHOT. Make sure the groupId, artifactId and version 
are spelled correctly and the relevant Maven repositories are configured.` I resolved that by adding settings.xml file with sonatype snapshots repo defined to be able to fetch the needed artfacts, I used https://github.com/quarkus-qe/quarkus-test-suite/blob/main/.github/quarkus-snapshots-mvn-settings.xml one.

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/ProjectInfoCommandHandler.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/ProjectInfoCommandHandler.java
@@ -3,10 +3,6 @@ package io.quarkus.devtools.commands.handlers;
 import static io.quarkus.devtools.commands.handlers.UpdateProjectCommandHandler.ITEM_FORMAT;
 import static io.quarkus.devtools.commands.handlers.UpdateProjectCommandHandler.printSeparator;
 import static io.quarkus.devtools.commands.handlers.UpdateProjectCommandHandler.updateInfo;
-import static io.quarkus.devtools.messagewriter.MessageFormatter.format;
-import static io.quarkus.devtools.messagewriter.MessageFormatter.Format.BOLD;
-import static io.quarkus.devtools.messagewriter.MessageFormatter.Format.GREEN;
-import static io.quarkus.devtools.messagewriter.MessageFormatter.Format.RED;
 import static io.quarkus.devtools.messagewriter.MessageIcons.UP_TO_DATE_ICON;
 import static io.quarkus.devtools.project.state.ProjectStates.resolveProjectState;
 
@@ -22,6 +18,7 @@ import io.quarkus.devtools.commands.ProjectInfo;
 import io.quarkus.devtools.commands.data.QuarkusCommandException;
 import io.quarkus.devtools.commands.data.QuarkusCommandInvocation;
 import io.quarkus.devtools.commands.data.QuarkusCommandOutcome;
+import io.quarkus.devtools.messagewriter.MessageFormatter;
 import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.devtools.project.state.ExtensionProvider;
 import io.quarkus.devtools.project.state.ModuleState;
@@ -74,9 +71,9 @@ public class ProjectInfoCommandHandler implements QuarkusCommandHandler {
         printSeparator(log);
 
         if (providerInfo.isEmpty()) {
-            log.info(format(RED, "No Quarkus platform BOMs found"));
+            log.info(MessageFormatter.red("No Quarkus platform BOMs found"));
         } else {
-            log.info(format(BOLD, "Quarkus platform BOMs:"));
+            log.info(MessageFormatter.bold("Quarkus platform BOMs:"));
             boolean recommendExtraImports = false;
             for (PlatformInfo platform : providerInfo.values()) {
                 if (!platform.isImported()) {
@@ -87,7 +84,8 @@ public class ProjectInfoCommandHandler implements QuarkusCommandHandler {
                 if (platform.getRecommended() == null) {
                     sb.append(
                             String.format(ITEM_FORMAT, "-",
-                                    "[" + format(RED, platform.getImported().toCompactCoords()) + "] not recommended"));
+                                    "[" + MessageFormatter.red(platform.getImported().toCompactCoords())
+                                            + "] not recommended"));
                     recommendationsAvailable = true;
                 } else if (platform.isVersionUpdateRecommended()) {
                     sb.append(String.format(ITEM_FORMAT, "~",
@@ -103,7 +101,7 @@ public class ProjectInfoCommandHandler implements QuarkusCommandHandler {
                 for (PlatformInfo platform : providerInfo.values()) {
                     if (platform.getImported() == null) {
                         log.info(String.format(ITEM_FORMAT, "+",
-                                "[ " + format(GREEN, platform.getImported().toCompactCoords()) + "]"));
+                                "[ " + MessageFormatter.green(platform.getImported().toCompactCoords()) + "]"));
                     }
                 }
                 recommendationsAvailable = true;
@@ -141,8 +139,9 @@ public class ProjectInfoCommandHandler implements QuarkusCommandHandler {
         }
         if (recommendationsAvailable) {
             log.info(
-                    "Some version alignment recommendation are available and shown by '[" + format(RED, "current") + " -> "
-                            + format(GREEN, "recommended") + "]'.");
+                    "Some version alignment recommendation are available and shown by '[" + MessageFormatter.red("current")
+                            + " -> "
+                            + MessageFormatter.green("recommended") + "]'.");
         }
 
         printSeparator(log);
@@ -154,7 +153,7 @@ public class ProjectInfoCommandHandler implements QuarkusCommandHandler {
         if (provider.getExtensions().isEmpty()) {
             return recommendationsAvailable;
         }
-        log.info(format(BOLD, "Extensions from " + provider.getKey() + ":"));
+        log.info(MessageFormatter.bold("Extensions from " + provider.getKey() + ":"));
         final StringBuilder sb = new StringBuilder();
         for (TopExtensionDependency dep : provider.getExtensions()) {
             sb.setLength(0);
@@ -171,8 +170,9 @@ public class ProjectInfoCommandHandler implements QuarkusCommandHandler {
             if (dep.isNonRecommendedVersion()) {
                 recommendationsAvailable = true;
                 sb.append(String.format(ITEM_FORMAT, "-",
-                        dep.getArtifact().getKey().toGacString() + ":[" + format(RED, dep.getArtifact().getVersion()) + " -> "
-                                + format(GREEN, "managed") + "]"));
+                        dep.getArtifact().getKey().toGacString() + ":[" + MessageFormatter.red(dep.getArtifact().getVersion())
+                                + " -> "
+                                + MessageFormatter.green("managed") + "]"));
             } else {
                 sb.append(
                         String.format(ITEM_FORMAT, UP_TO_DATE_ICON.iconOrMessage(), dep.getArtifact().getKey().toGacString()));

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/UpdateProjectCommandHandler.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/UpdateProjectCommandHandler.java
@@ -196,7 +196,7 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
                 MessageFormatter.bold(
                         "Do you want to apply the generated update recipes with OpenRewrite?")
                 + " ([" + MessageFormatter.green("y") + "]es, [" + MessageFormatter.red("n") + "]o, ["
-                + MessageFormatter.blue("d") + "]ry-run)"
+                + MessageFormatter.blue("d") + "]ry-run + [Enter])"
                 + System.lineSeparator() + System.lineSeparator());
         try (Scanner scanner = new Scanner(new FilterInputStream(System.in) {
             @Override

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/UpdateProjectCommandHandler.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/UpdateProjectCommandHandler.java
@@ -276,18 +276,18 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
                                     + "]"));
                 }
             }
-            if (!platformUpdateInfo.getNewImports().isEmpty()) {
-                for (PlatformInfo importInfo : platformUpdateInfo.getNewImports()) {
-                    log.info(String.format(UpdateProjectCommandHandler.ITEM_FORMAT, "+",
-                            "[" + MessageFormatter.green(importInfo.getRecommended().toCompactCoords()) + "] to add"));
-                }
-            }
             if (platformUpdateInfo.isImportsToBeRemoved()) {
                 for (PlatformInfo importInfo : platformUpdateInfo.getPlatformImports().values()) {
                     if (importInfo.getRecommended() == null) {
                         log.info(String.format(UpdateProjectCommandHandler.ITEM_FORMAT, "-",
-                                "[" + MessageFormatter.red(importInfo.getImported().toCompactCoords()) + "] to remove"));
+                                "[" + MessageFormatter.red(importInfo.getImported().toCompactCoords()) + "]"));
                     }
+                }
+            }
+            if (!platformUpdateInfo.getNewImports().isEmpty()) {
+                for (PlatformInfo importInfo : platformUpdateInfo.getNewImports()) {
+                    log.info(String.format(UpdateProjectCommandHandler.ITEM_FORMAT, "+",
+                            "[" + MessageFormatter.green(importInfo.getRecommended().toCompactCoords()) + "]"));
                 }
             }
             log.info("");

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/UpdateProjectCommandHandler.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/UpdateProjectCommandHandler.java
@@ -1,11 +1,5 @@
 package io.quarkus.devtools.commands.handlers;
 
-import static io.quarkus.devtools.messagewriter.MessageFormatter.format;
-import static io.quarkus.devtools.messagewriter.MessageFormatter.Format.BLUE;
-import static io.quarkus.devtools.messagewriter.MessageFormatter.Format.BOLD;
-import static io.quarkus.devtools.messagewriter.MessageFormatter.Format.GREEN;
-import static io.quarkus.devtools.messagewriter.MessageFormatter.Format.RED;
-import static io.quarkus.devtools.messagewriter.MessageFormatter.Format.UNDERLINE;
 import static io.quarkus.devtools.messagewriter.MessageIcons.UP_TO_DATE_ICON;
 import static io.quarkus.devtools.project.state.ProjectStates.resolveProjectState;
 import static io.quarkus.devtools.project.update.ProjectUpdateInfos.resolvePlatformUpdateInfo;
@@ -31,6 +25,7 @@ import io.quarkus.devtools.commands.UpdateProject;
 import io.quarkus.devtools.commands.data.QuarkusCommandException;
 import io.quarkus.devtools.commands.data.QuarkusCommandInvocation;
 import io.quarkus.devtools.commands.data.QuarkusCommandOutcome;
+import io.quarkus.devtools.messagewriter.MessageFormatter;
 import io.quarkus.devtools.messagewriter.MessageIcons;
 import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.devtools.project.BuildTool;
@@ -129,9 +124,8 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
                         additionalUpdateRecipes, request);
                 quarkusProject.log().info("");
                 quarkusProject.log()
-                        .info("We have generated a recipe file to update your project (versions updates + specific recipes):");
-                quarkusProject.log().info(format(UNDERLINE,
-                        projectDir.relativize(recipe).toString()));
+                        .info("We have generated a recipe file to update your project (version updates + specific recipes):");
+                quarkusProject.log().info(MessageFormatter.underline(projectDir.relativize(recipe).toString()));
                 quarkusProject.log().info("");
 
                 if (rewriteDryRun) {
@@ -181,8 +175,7 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
                     final Path patchFile = rewriteDir.resolve("rewrite.patch");
                     if (rewriteDryRun && Files.isRegularFile(patchFile)) {
                         quarkusProject.log().info("Patch file available:");
-                        quarkusProject.log().info(format(UNDERLINE,
-                                projectDir.relativize(patchFile).toString()));
+                        quarkusProject.log().info(MessageFormatter.underline(projectDir.relativize(patchFile).toString()));
                         quarkusProject.log().info("");
                     }
 
@@ -200,9 +193,10 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
 
     private static String askUserConfirmationForUpdate(MessageWriter log) {
         System.out.print(System.lineSeparator() +
-                format(BOLD,
-                        "Do you want run the generated update recipe with OpenRewrite ?")
-                + " ([" + format(GREEN, "y") + "]es, [" + format(RED, "n") + "]o, [" + format(BLUE, "d") + "]ry-run)"
+                MessageFormatter.bold(
+                        "Do you want to apply the generated update recipes with OpenRewrite?")
+                + " ([" + MessageFormatter.green("y") + "]es, [" + MessageFormatter.red("n") + "]o, ["
+                + MessageFormatter.blue("d") + "]ry-run)"
                 + System.lineSeparator() + System.lineSeparator());
         try (Scanner scanner = new Scanner(new FilterInputStream(System.in) {
             @Override
@@ -212,7 +206,7 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
         })) {
             return scanner.nextLine();
         } catch (Exception e) {
-            log.debug("Failed to collect user input for analytics", e);
+            log.debug("Unable to get user confirmation", e);
             return "";
         }
     }
@@ -256,7 +250,7 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
             MessageWriter log) {
         printSeparator(log);
         if (currentState.getPlatformBoms().isEmpty()) {
-            log.info(format(RED, "The project does not import any Quarkus platform BOM"));
+            log.info(MessageFormatter.red("The project does not import any Quarkus platform BOM"));
             printSeparator(log);
             return false;
         }
@@ -266,43 +260,44 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
             return false;
         }
         if (currentState == recommendedState) {
-            log.info(format(GREEN, "The project is up-to-date)"));
+            log.info(MessageFormatter.green("The project is up-to-date)"));
             printSeparator(log);
             return false;
         }
 
         if (platformUpdateInfo.isPlatformUpdatesAvailable()) {
-            log.info(format(BOLD, "Suggested Quarkus platform BOM updates:"));
+            log.info(MessageFormatter.bold("Suggested Quarkus platform BOM updates:"));
             if (!platformUpdateInfo.getImportVersionUpdates().isEmpty()) {
                 for (PlatformInfo importInfo : platformUpdateInfo.getImportVersionUpdates()) {
                     log.info(String.format(UpdateProjectCommandHandler.ITEM_FORMAT, "~",
                             importInfo.getImported().getKey().toGacString() + ":pom:["
-                                    + format(RED, importInfo.getImported().getVersion()) + " -> "
-                                    + format(GREEN, importInfo.getRecommendedVersion())
+                                    + MessageFormatter.red(importInfo.getImported().getVersion()) + " -> "
+                                    + MessageFormatter.green(importInfo.getRecommendedVersion())
                                     + "]"));
                 }
             }
             if (!platformUpdateInfo.getNewImports().isEmpty()) {
                 for (PlatformInfo importInfo : platformUpdateInfo.getNewImports()) {
                     log.info(String.format(UpdateProjectCommandHandler.ITEM_FORMAT, "+",
-                            "[" + format(GREEN, importInfo.getRecommended().toCompactCoords()) + "] to add"));
+                            "[" + MessageFormatter.green(importInfo.getRecommended().toCompactCoords()) + "] to add"));
                 }
             }
             if (platformUpdateInfo.isImportsToBeRemoved()) {
                 for (PlatformInfo importInfo : platformUpdateInfo.getPlatformImports().values()) {
                     if (importInfo.getRecommended() == null) {
                         log.info(String.format(UpdateProjectCommandHandler.ITEM_FORMAT, "-",
-                                "[" + format(RED, importInfo.getImported().toCompactCoords()) + "] to remove"));
+                                "[" + MessageFormatter.red(importInfo.getImported().toCompactCoords()) + "] to remove"));
                     }
                 }
             }
             log.info("");
         } else if (!extensionsUpdateInfo.shouldUpdateExtensions()) {
-            log.info(format(GREEN, "The project is up-to-date " + MessageIcons.UP_TO_DATE_ICON.iconOrMessage()));
+            log.info(MessageFormatter.green("The project is up-to-date " + MessageIcons.UP_TO_DATE_ICON.iconOrMessage()));
             printSeparator(log);
             return false;
         } else {
-            log.info(format(GREEN, "Quarkus platform BOM(s) are up-to-date " + MessageIcons.UP_TO_DATE_ICON.iconOrMessage()));
+            log.info(MessageFormatter
+                    .green("Quarkus platform BOMs are up-to-date " + MessageIcons.UP_TO_DATE_ICON.iconOrMessage()));
             log.info("");
         }
 
@@ -321,7 +316,7 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
             if (extensions.isEmpty()) {
                 continue;
             }
-            log.info(format(BOLD,
+            log.info(MessageFormatter.bold(
                     "Suggested extensions updates for '%s':".formatted(platform.getRecommendedProviderKey())));
             for (ExtensionUpdateInfo e : extensions) {
 
@@ -340,13 +335,14 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
                         case RECOMMEND_PLATFORM_MANAGED:
                             log.info(String.format(UpdateProjectCommandHandler.ITEM_FORMAT, "-",
                                     e.getCurrentDep().getArtifact().getKey().toGacString()) + ":["
-                                    + format(RED, e.getCurrentDep().getVersion()) + " -> " + format(GREEN, "managed") + "]");
+                                    + MessageFormatter.red(e.getCurrentDep().getVersion()) + " -> "
+                                    + MessageFormatter.green("managed") + "]");
                             break;
                         case ADD_VERSION:
                             log.info(String.format(UpdateProjectCommandHandler.ITEM_FORMAT, versionUpdateType.equals(
                                     ExtensionUpdateInfo.VersionUpdateType.ADD_VERSION) ? "+" : "-",
                                     e.getCurrentDep().getArtifact().getKey().toGacString()) + ":["
-                                    + format(RED, "managed") + " -> " + format(GREEN,
+                                    + MessageFormatter.red("managed") + " -> " + MessageFormatter.green(
                                             e.getRecommendedDependency().getVersion())
                                     + "]");
                             break;
@@ -362,12 +358,12 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
         final List<ExtensionUpdateInfo> simpleVersionUpdates = extensionsUpdateInfo.getSimpleVersionUpdates().stream()
                 .filter(u -> !u.getCurrentDep().isPlatformExtension()).toList();
         if (!simpleVersionUpdates.isEmpty()) {
-            log.info(format(BOLD, "Suggested extensions updates from other origins:"));
+            log.info(MessageFormatter.bold("Suggested extension updates from other origins:"));
             for (ExtensionUpdateInfo u : simpleVersionUpdates) {
                 if (u.getVersionUpdateType() == ExtensionUpdateInfo.VersionUpdateType.PLATFORM_MANAGED) {
                     log.info(String.format(UpdateProjectCommandHandler.ITEM_FORMAT, "-",
                             u.getCurrentDep().getArtifact().getKey().toGacString()) + ":["
-                            + format(RED, u.getCurrentDep().getVersion()) + "]");
+                            + MessageFormatter.red(u.getCurrentDep().getVersion()) + "]");
                 } else {
                     log.info(String.format(UpdateProjectCommandHandler.ITEM_FORMAT, "~",
                             updateInfo(u)));
@@ -390,9 +386,9 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
     }
 
     static String updateInfo(ArtifactCoords current, String newVersion) {
-        return current.getKey().toGacString() + ":[" + format(RED,
+        return current.getKey().toGacString() + ":[" + MessageFormatter.red(
                 current.getVersion()) + " -> "
-                + format(GREEN, newVersion) + "]";
+                + MessageFormatter.green(newVersion) + "]";
     }
 
     @SuppressWarnings({ "rawtypes", "unchecked" })

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/update/rewrite/QuarkusUpdateCommand.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/update/rewrite/QuarkusUpdateCommand.java
@@ -4,6 +4,7 @@ import static io.quarkus.devtools.project.update.rewrite.QuarkusUpdateRecipe.REC
 
 import java.io.BufferedReader;
 import java.io.File;
+import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.nio.charset.StandardCharsets;
@@ -155,7 +156,6 @@ public class QuarkusUpdateCommand {
         command.add(mvnBinary);
         command.add("-B");
         command.add("-e");
-        command.add("clean");
         command.add("process-sources");
         final String mavenSettings = getMavenSettingsArg();
         if (mavenSettings != null) {
@@ -195,10 +195,23 @@ public class QuarkusUpdateCommand {
 
         log.info("");
 
-        String logInfo = logFile != null ? "Logs can be found at: %s".formatted(baseDir.relativize(logFile).toString())
-                : "See the execution logs above for more details";
+        String logInfo;
+
+        if (logFile != null) {
+            logInfo = "Logs can be found at: %s".formatted(baseDir.relativize(logFile).toString());
+
+            try {
+                // we delete the log file prior to executing the command so that we start clean
+                Files.deleteIfExists(logFile);
+            } catch (IOException e) {
+                // ignore, it's not a big deal
+            }
+        } else {
+            logInfo = "See the execution logs above for more details";
+        }
 
         log.info("Update in progress (this may take a while):");
+
         for (Map.Entry<String, List<String>> e : commands.entrySet()) {
             log.info("  - executing " + e.getKey() + " command...");
             execute(e.getKey(), e.getValue(), log, logFile, logInfo);
@@ -218,7 +231,7 @@ public class QuarkusUpdateCommand {
 
         try {
             if (logFile != null) {
-                Files.writeString(logFile, "\n\n\nRunning: " + String.join(" ", effectiveCommand) + "\n",
+                Files.writeString(logFile, "Running: " + String.join(" ", effectiveCommand) + "\n",
                         StandardOpenOption.APPEND, StandardOpenOption.CREATE);
             }
 
@@ -260,13 +273,16 @@ public class QuarkusUpdateCommand {
                 }
 
                 log.info("");
-
             }
 
             int exitCode = process.waitFor();
             if (exitCode != 0) {
                 log.info("");
                 throw new QuarkusUpdateExitErrorException("The %s command exited with an error. %s".formatted(name, logInfo));
+            }
+
+            if (logFile != null) {
+                Files.writeString(logFile, "\n\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n\n");
             }
         } catch (QuarkusUpdateException e) {
             throw e;

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/update/rewrite/QuarkusUpdatesRepository.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/update/rewrite/QuarkusUpdatesRepository.java
@@ -1,7 +1,5 @@
 package io.quarkus.devtools.project.update.rewrite;
 
-import static io.quarkus.devtools.messagewriter.MessageFormatter.Format.GREEN;
-
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.UncheckedIOException;
@@ -106,14 +104,15 @@ public final class QuarkusUpdatesRepository {
                 }
                 if (log.isDebugEnabled()) {
                     log.debug(String.format(
-                            "=> %d specific recipe(s) found (compatible with OpenRewrite %s plugin version: %s)",
+                            "=> %d specific recipe" + (recipes.size() != 1 ? "s" : "")
+                                    + " found (compatible with OpenRewrite %s plugin version: %s)",
                             recipes.size(),
                             buildTool,
                             propRewritePluginVersion));
                 } else {
                     log.info(String.format(
-                            "=> %s specific recipe(s) found",
-                            MessageFormatter.format(GREEN, String.valueOf(recipes.size()))));
+                            "=> %s specific recipe" + (recipes.size() != 1 ? "s" : "") + " found",
+                            MessageFormatter.green(String.valueOf(recipes.size()))));
                 }
 
             } catch (BootstrapMavenException e) {

--- a/independent-projects/tools/message-writer/src/main/java/io/quarkus/devtools/messagewriter/MessageFormatter.java
+++ b/independent-projects/tools/message-writer/src/main/java/io/quarkus/devtools/messagewriter/MessageFormatter.java
@@ -2,11 +2,11 @@ package io.quarkus.devtools.messagewriter;
 
 public class MessageFormatter {
 
-    public enum Format {
+    enum Format {
         RED("\u001B[91m"),
         GREEN("\u001b[32m"),
         BLUE("\u001b[94m"),
-        RESET("\u001b[39m"),
+        RESET_COLOR("\u001b[39m"),
         UNDERLINE("\u001b[4m"),
         NO_UNDERLINE("\u001b[24m"),
         BOLD("\u001b[1m"),
@@ -24,8 +24,23 @@ public class MessageFormatter {
         }
     }
 
-    public static String format(Format format, String text) {
-        return format + text + Format.RESET;
+    public static String bold(String text) {
+        return Format.BOLD + text + Format.NO_BOLD;
     }
 
+    public static String underline(String text) {
+        return Format.UNDERLINE + text + Format.NO_UNDERLINE;
+    }
+
+    public static String red(String text) {
+        return Format.RED + text + Format.RESET_COLOR;
+    }
+
+    public static String green(String text) {
+        return Format.GREEN + text + Format.RESET_COLOR;
+    }
+
+    public static String blue(String text) {
+        return Format.BLUE + text + Format.RESET_COLOR;
+    }
 }

